src/libraries/System.Text.Json/src/ILLink/ILLink.Substitutions.xml

```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Text.Json">
+    <type fullname="System.Text.Json.AppContextSwitchHelper">
+      <method signature="System.Boolean get_IsDefaultReflectionDisabled()" body="stub" value="true" feature="System.Text.Json.Serialization.DisableDefaultReflection" featurevalue="true"/>
```

Comment:
We avoid `Disable` when naming the feature flags. `Disable` in the names of the feature flags leads to double-negation that is harder to reason about.

It is fine to have the flag to be true by default. We have many flags like that.

Reply:
Note that if/when we decide to make this a public API, we should use similar names.

For example `System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported` is the name of the public API, and the backing feature switch is named `System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported`.

https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeFeature.NonNativeAot.cs#L4-L14

This makes it really easy to know what feature switch maps to what API. In an ideal world, these would always be the same. But we can't do that for everything. My suggestion would be to consider what name we would use in a C# API, and then use the same for the feature switch.

Reply:
We settled on `UseReflectionDefault` with a default value of `true`. I'm guessing this could eventually map to a `JsonSerializer.UseReflectionDefault` static property should the need for such a property arise.